### PR TITLE
[AC-5110] - Update accelerate to use SSO

### PR DESCRIPTION
**Pull request by yotammanor**
Changes Introduced in [AC-5110](https://masschallenge.atlassian.net/browse/AC-5110):

- Configure env vars to work with accelerate.
- Fix a few dev env issues

**Test:**
**IMPORTANT NOTE: If we're doing something wrong, we risk either breaking down our login process, or exposing our user data and credentials. Please test this very carefully.**

- See accelerate for functional testing.
- If you run `make delete-vms;make build;make run-server` on development, it will run forever and fail. On branch, it does not.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '2.3'
 services:
   mysql:
-    image: mysql
+    image: mysql:5.7.21
```
Comment (pbx) [inline docker-compose.yml:4]: Are we ready for 5.7? I'm not up to date on our plans on this.  I just know that Accelerate production is still at 5.5 and that [AC-4618](https://masschallenge.atlassian.net/browse/AC-4618) is still TODO.

**Review by pbx — CHANGES_REQUESTED** [review]
My more detailed view is on the accelerate PR, but in short: **SSO on test3 didn't work for me**. I was able to log in via test3.masschallenge.org, but test3.api.masschallenge.org did not see me as authenticated.

File: docker-compose.yml (hunk 1 above)
Comment (yotammanor) [reply docker-compose.yml:4]: @pbx we are already on 5.7, run `make db-shell` on development and you'll see.
I believe that without this lock the mysql installed version is 8, and we're not ready for _that_ one.

Comment (yotammanor) [reply docker-compose.yml:4]: @pbx we are not using mysql container on prod, this is only a development environment configuration at the moment. This PR doesn't change that (and again, we were already on 5.7 locally before this PR).

**Review by pbx — APPROVED** [review]

File: .dev.env [hunk 2]
```diff
@@ -4,7 +4,8 @@ DJANGO_HIREDIS_CACHE_LOCATION=redis:6379
 DJANGO_ALLOWED_HOST=""
 DATABASE_URL=mysql://root:root@mysql/mc_dev
 EMAIL_URL=console://
-JWT_SECRET_KEY=ffffffff-ffff-ffff-ffff-ffffffffffffjwt
+JWT_AUTH_COOKIE=jwt
```
Comment (shanbady) [inline .dev.env:7]: can we make this the same name as what we use in accelerate? (JWT_COOKIE_NAME)

File: assets/gulpfile.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ gulp.task('build_js', function() {
       }
     }))
     .pipe(addsrc.prepend(['/gulp/js/dev/lib/*.js', '!/gulp/js/dev/lib/jquery.min.js']))
-    .pipe(addsrc.prepend('/gulp/js/dev/lib/jquery.min.js'))
+    .pipe(addsrc.prepend('/gulp/js/dev/lib/jquery.min.js', { allowEmpty: true }))
```
Comment (shanbady) [inline assets/gulpfile.js:56]: why was the allowEmpty necessary?

**Review by shanbady — CHANGES_REQUESTED** [review]

**Review by shanbady — CHANGES_REQUESTED** [review]
only blocking comment is about the cookie name. will look tomorrow morning and get this merged.

**Review by shanbady — APPROVED** [review]

File: .dev.env (hunk 2 above)
Comment (yotammanor) [reply .dev.env:7]: I was debating whether this should be the same as accelerate or the same as drf-jwt's name. We can pick that up later.

File: assets/gulpfile.js (hunk 3 above)
Comment (yotammanor) [reply assets/gulpfile.js:56]: The assets build step failed otherwise. try it and you'll see.

**Pull request merged**